routes/web.php

```diff
@@ -11,24 +11,35 @@
 */
 
 
-Route::get('/', function () {
-    return view('welcome');
-})->middleware('guest');
+// Route::get('/', function () {
+//     return view('welcome');
+// })->middleware('guest');
+
+
+Route::get('/','HomeController@show' );
 
 // remove later
 Route::get('/upload', function () {
     return view('modules.upload');
 });
 
-Auth::routes();
 
-Route::get('/dashboard', 'DashboardController@index')->name('dashboard');
+Auth::routes(); // includes routes for login, register, forget password
+
+// only authenticated user can access this route group
+Route::group(['prefix'=>'dashboard', 'middleware' => 'auth'], function () {
```

Comment:
Actually in each corresponding controller I have defined the auth middleware. I think whichever way is OK for now. But, I'd like to keep the middleware definition in just one place, so it's not confusing. It's either here in the routes definition, or in each controller. I don't want to mix it up.

For now, we can do this because none of the controllers require a special middleware. But if we do, we'll have to choose between specifying the middleware in controllers or here.